src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -508,6 +513,51 @@ private void validatePluginArguments() {
         // finally, give each plugin a chance to trim down any unseen instances from it's own list
         pluginDescriptors.entrySet().forEach(e -> e.getValue().validateArguments());
     }
+    /**
+     * Check the provided value against any range constraints specified in the Argument annotation
+     * for the corresponding field. Throw an exception if limits are violated.
+     *
+     * - Only checks numeric types (int, double, etc.)
+     */
+    private void checkArgumentRange(final ArgumentDefinition argumentDefinition, final Object argumentValue) {
+        // Only validate numeric types
```

Comment:
Please add a comment here saying that we only check numeric types because we have already ensured that only numeric types have bounds.